posts/20230223/index.qmd

```diff
@@ -0,0 +1,67 @@
+
+#### Data
+The data should be encoded properly, with categorical data like parid (property identifier), nbhd (neighborhood code), yrblt (year built), zip21 (ZIP code of area), sale_date (sale date), and luc (property class) treated as such. Furthermore, month and year variables should be extracted and matched to the socio-economic data matching it on those variables where possible. 
+
+![Figure 1. Another architecture for how the authors could have approached their high-price/low-price classificaiton problem; compare to the authors' architecture in Figure 9 of the paper.](alternative_approach.png)
+
+#### Model Training
+Before training the model, we should have an understanding about what exactly we want the model to do. Looking at Fig. 2 in the paper, we can see that the occurrence of high price instances surpassed low price instances in the last few years; thus it would probably behoove the authors to build a model that is optimized to predict high-price (Class = 1) outcomes to apply this model to new data. 
+
+In order to train the model, the authors should consider the time-sensitive deviation in high-price classes in the data by using a time series split of the data in their 10-fold cross validation. Furthermore, we propose to keep 20% of the latest data as the ultimate holdout test for evaluating the winning model. For each fold of the time-split data, the different models of interest (i.e. Random Forest, XGBoost, Logistic Regression) can be trained and hyperparameter tuned. 
+
+#### Model Appraisal
+Because in this alternative approach we have decided that detecting high-price outcomes is more valuable, and that the distribution of high- and low-price though balanced, differs over time, the main method by which we can evaluate the best performing classifier is through the metric of F1, which is a function of the metrics of precision and recall. The authors can then find the best F1 score for a given model and its respective optimal hyperparameters. Then, the authors can validate the model with the best F1 score on the holdout test to see how it performs on the latest data. They can also evaluate what features of the model contributed to discriminating amongst the high and low-price classes. 
+
+With this alternative framework, the authors would be better able to rectify some of the glaring procedural errors in this paper, namely the huge amounts of data leakage that went into every layer of the authors’ research process (see Fig. 9) in this paper. In this way, they would be able to be more confident about which models performed better at discriminating high-price homes and be better able to understand what features of the data contributed to the model’s performance.
```

Comment:
I think the another fundamental question to answer is what the point of model comparison is, and why we compare this particular set of models.

Reply:
See above

Reply:
I deleted this section as I've gotten rid of the alternative approach